### PR TITLE
HomeWork-jv-array-list-14-of-april

### PR DESCRIPTION
**Pull request by ViktoriiaSvidovska**

### CODE REVIEW

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -1,48 +1,103 @@
...
+        }
+        size += list.size();
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index: " + index);
+        }
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ArrayList.java:48]: Move this to the separate method, like "checkIndex"

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -1,48 +1,103 @@
...
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        ensureCapacity(size + list.size());
+        for (int i = 0; i < list.size(); i++) {
+            elements[size + i] = list.get(i);
+        }
+        size += list.size();
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ArrayList.java:41]:
```suggestion
        for (int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
```

File: src/main/java/core/basesyntax/ArrayList.java [hunk 3]
```diff
@@ -1,48 +1,103 @@
...
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index: " + index);
+        }
+        final T removed = elements[index];
+        System.arraycopy(elements, index + 1, elements, index, size - index - 1);
+        elements[size - 1] = null;
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ArrayList.java:67]:
```suggestion
```

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]
Good job!

**Pull request closed without merging**